version/string.go

```diff
@@ -9,32 +9,40 @@ import (
 	"strings"
 )
 
-var (
-	// String is displayed when CLI arg --version is used
-	String string
+// GitCommit is set in the build script at compile time
+var GitCommit string
 
-	// GitCommit is set in the build script at compile time
-	GitCommit string
-)
+// Versions contains the versions relevant to a build of avalanchego. In
+// addition to supporting construction of the strings displayed by
+// --version and --version-json, it can be used to unmarshal the output
+// of --version-json.
+type Versions struct {
+	Application string `json:"application"`
+	Database    string `json:"database"`
+	RPCChainVM  uint64 `json:"rpcchainvm"`
+	// Commit may be empty if GitCommit was not set at compile time
+	Commit string `json:"commit"`
+	Go     string `json:"go"`
+}
 
-func init() {
-	format := "%s [database=%s, rpcchainvm=%d"
-	args := []interface{}{
-		CurrentApp,
-		CurrentDatabase,
-		RPCChainVMProtocol,
+func GetVersions() *Versions {
+	versions := &Versions{
+		Application: CurrentApp.String(),
+		Database:    CurrentDatabase.String(),
+		RPCChainVM:  uint64(RPCChainVMProtocol),
+		Go:          strings.TrimPrefix(runtime.Version(), "go"),
 	}
 	if GitCommit != "" {
-		format += ", commit=%s"
-		args = append(args, GitCommit)
+		versions.Commit = GitCommit
```

Comment:
The default value of `versions.Commit` is `""`. So I think we can just directly set this in the construction of the struct rather than checking if it is populated here.